x/ccv/provider/keeper/validator_set_storage.go

```diff
@@ -0,0 +1,114 @@
+package keeper
+
+import (
+	"fmt"
+
+	"cosmossdk.io/math"
+	storetypes "cosmossdk.io/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/interchain-security/v5/x/ccv/provider/types"
+)
+
+// getValidatorKey constructs the key to access a given validator, stored under a given prefix.
+func (k Keeper) getValidatorKey(prefix []byte, providerAddr types.ProviderConsAddress) []byte {
+	return append(prefix, providerAddr.ToSdkConsAddr()...)
+}
+
+// setValidator stores the given `validator` in the validator set stored under the given prefix.
+func (k Keeper) setValidator(
+	ctx sdk.Context,
+	prefix []byte,
+	validator types.ConsensusValidator,
+) error {
+	store := ctx.KVStore(k.storeKey)
+	bz, err := validator.Marshal()
+	if err != nil {
+		return fmt.Errorf("marshalling ConsumerValidator: %w", err)
+	}
+
+	store.Set(k.getValidatorKey(prefix, types.NewProviderConsAddress(validator.ProviderConsAddr)), bz)
+	return nil
+}
+
+// setValSet resets the validator set stored under the given prefix to the provided `nextValidators`.
+func (k Keeper) setValSet(ctx sdk.Context, prefix []byte, nextValidators []types.ConsensusValidator) error {
+	k.deleteValSet(ctx, prefix)
+	for _, val := range nextValidators {
+		err := k.setValidator(ctx, prefix, val)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// deleteValidator removes validator with `providerAddr` address from the
+// validator set stored under the given prefix.
+func (k Keeper) deleteValidator(
+	ctx sdk.Context,
+	prefix []byte,
+	providerConsAddr types.ProviderConsAddress,
+) {
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(k.getValidatorKey(prefix, providerConsAddr))
+}
+
+// deleteValSet deletes all the stored consumer validators under the given prefix.
+func (k Keeper) deleteValSet(
+	ctx sdk.Context,
+	prefix []byte,
+) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := storetypes.KVStorePrefixIterator(store, prefix)
+
+	var keysToDel [][]byte
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		keysToDel = append(keysToDel, iterator.Key())
+	}
+	for _, delKey := range keysToDel {
+		store.Delete(delKey)
+	}
+}
+
+// isValidator returns `true` if the validator with `providerAddr` exists
```

Comment:
Previously you used `providerConsAddr`, while here `providerAddr`. Maybe make those consistent across.

Reply:
Made it consistently providerConsAddr in this file